4_predict/src/select_features.R

```diff
@@ -0,0 +1,273 @@
+drop_high_corr_attrs <- function(features, drop_columns = NULL, drop_exact = NULL,
+                                 dynamic_contains,
+                                 threshold_corr, first_drop_var_prefix = NULL,
+                                 cor_method){
+  #' 
+  #' @description Function to remove attributes that are undesired or are 
+  #' highly correlated with other attributes. 
+  #'
+  #' @param features table of features (columns) for each prediction point (rows).
+  #' @param drop_columns character vector of column names to remove from features. 
+  #' This is passed to dplyr::contains()
+  #' @param drop_exact character vector of exact column names to drop.
+  #' This is passed to dplyr::select()
```

Comment:
The difference is that `drop_exact` uses `select()` to remove the column, which requires the exact column name, and `drop_columns` uses `contains()` to remove the column, which matches all column names with a particular pattern. I can rename these args to be more helpful, though:
`drop_columns_select`, and `drop_columns_contains`

Reply:
I see the difference between these options, but agree with Galen that having both options potentially makes the `drop_high_corr_attrs()` function harder to grasp. You could consider just using `drop_columns_contains` and then an exact string should still work with `dplyr::contains()`. I don't have a strong opinion on that, but this thread reminds me that I've sometimes been surprised by unexpected string matches when typing out options like those currently included in `drop_columns`. It might be a nice feature to print a message that confirms which columns are dropped as long as the output isn't so verbose that a user is likely to just ignore it altogether (here's an [example](https://github.com/USGS-R/ds-pipelines-targets-example-wqp/blob/main/1_inventory/src/check_characteristics.R#L77-L79) of that idea, but with a warning message).

Reply:
Thanks, @jds485 - that's a good example of this sort of edge case.